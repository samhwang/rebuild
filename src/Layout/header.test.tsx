import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { BrowserRouter } from 'react-router-dom';
import Header from './header';

describe('Header', () => {
  it('Should render without crashing and match snapshot', () => {
    const tree = render(
      <BrowserRouter>
        <Header opened toggleOpened={() => {}} />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });

  it('Should be hidden when pass opened false', () => {
    const tree = render(
      <BrowserRouter>
        <Header opened={false} toggleOpened={() => {}} />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });

  it('Should change state when clicked on the burger button', () => {
    let opened = false;
    const toggle = () => {
      opened = true;
    };
    render(
      <BrowserRouter>
        <Header opened={opened} toggleOpened={toggle} />
      </BrowserRouter>
    );
    const burgerButton = screen.getByLabelText('BurgerButton');
    fireEvent.click(burgerButton);
    expect(opened).toEqual(true);
  });
});
