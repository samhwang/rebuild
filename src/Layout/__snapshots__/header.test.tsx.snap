// Vitest Snapshot v1

exports[`Header > Should be hidden when pass opened false 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="mantine-Header-root mantine-MediaQuery-media mantine-1e144hv"
      >
        <div
          style="display: flex; align-items: center; height: 100%;"
        >
          <button
            aria-label="BurgerButton"
            class="mantine-UnstyledButton-root mantine-Burger-root mantine-s1uxp1"
            type="button"
          >
            <div
              class="mantine-10gf2v3 mantine-Burger-burger"
            />
          </button>
          <div
            class="mantine-Text-root mantine-1pbxw0k"
          >
            Header Text
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="mantine-Header-root mantine-MediaQuery-media mantine-1e144hv"
    >
      <div
        style="display: flex; align-items: center; height: 100%;"
      >
        <button
          aria-label="BurgerButton"
          class="mantine-UnstyledButton-root mantine-Burger-root mantine-s1uxp1"
          type="button"
        >
          <div
            class="mantine-10gf2v3 mantine-Burger-burger"
          />
        </button>
        <div
          class="mantine-Text-root mantine-1pbxw0k"
        >
          Header Text
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header > Should render without crashing and match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="mantine-Header-root mantine-MediaQuery-media mantine-1e144hv"
      >
        <div
          style="display: flex; align-items: center; height: 100%;"
        >
          <button
            aria-label="BurgerButton"
            class="mantine-UnstyledButton-root mantine-Burger-root mantine-s1uxp1"
            type="button"
          >
            <div
              class="mantine-10gf2v3 mantine-Burger-burger"
              data-opened="true"
            />
          </button>
          <div
            class="mantine-Text-root mantine-1pbxw0k"
          >
            Header Text
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="mantine-Header-root mantine-MediaQuery-media mantine-1e144hv"
    >
      <div
        style="display: flex; align-items: center; height: 100%;"
      >
        <button
          aria-label="BurgerButton"
          class="mantine-UnstyledButton-root mantine-Burger-root mantine-s1uxp1"
          type="button"
        >
          <div
            class="mantine-10gf2v3 mantine-Burger-burger"
            data-opened="true"
          />
        </button>
        <div
          class="mantine-Text-root mantine-1pbxw0k"
        >
          Header Text
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
